lib/types/number.js

```diff
@@ -312,10 +312,16 @@ internals.normalizeExponent = function (str) {
 internals.normalizeDecimal = function (str) {
 
     str = str
+        // Remove leading plus signs
         .replace(/^\+/, '')
+        // Remove trailing zeros if there is a decimal point
         .replace(/\.0+$/, '')
+        // Remove a trailing decimal point
+        .replace(/\.$/, '')
```

Comment:
This is redundant if you change the ‘+’ in the previous line with a ‘*’.

Reply:
Agreed. I'm removing this lines and replacing the regex in the previous to also remove the trailing decimal point.